language/tools/vm-genesis/src/lib.rs

```diff
@@ -144,35 +144,38 @@ fn create_and_initialize_main_accounts(
             Value::transaction_argument_signer_reference(fee_account_address),
             Value::transaction_argument_signer_reference(tc_account_address),
             Value::address(tc_account_address),
-            Value::vector_u8(AuthenticationKey::prefix(&genesis_auth_key).to_vec()), // TODO: different key here?
-            Value::vector_u8(genesis_auth_key.to_vec()),
             Value::vector_u8(genesis_auth_key.to_vec()),
         ],
     );
 
-    // TODO: use signer to eliminate this
     context.set_sender(config_address());
     context.exec(
         "LibraAccount",
         "rotate_authentication_key",
         vec![],
-        vec![Value::vector_u8(genesis_auth_key.to_vec())],
+        vec![
+            Value::transaction_argument_signer_reference(config_address()),
+            Value::vector_u8(genesis_auth_key.to_vec()),
+        ],
```

Comment:
I think this should be able to be done in the main genesis procedure now that `rotate_authentication_key` takes a `signer`.

Reply:
good call!